### PR TITLE
refactor: separate out webpacking for tests

### PR DESCRIPTION
**Pull request by natenichols**
## High Level Overview of Change
`npm i` was taking a long time to run because of webpacking browser tests. This splits building browser tests into its own script.

### Context of Change
Previously, we were webpacking everything in one command. This meant that `npm i` took a long time to run, because in addition to webpacking `xrpl`, it also had to webpack browser tests. This PR separates out the webpacking into two commands `npm run build:browserTests`, which builds the browser tests, and `npm build:web`, which builds the `xrpl` library. `npm run build:web` runs on installation, and `npm run build:browserTests` only runs when `test:browser` script is run.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

### CODE REVIEW

File: test/webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    cache: true,
+    externals: [
+      {
+        'xrpl-local': 'xrpl',
+        net: 'null',
+      },
+    ],
+    entry: testFileName,
+    output: {
+      library: match[1].replace(/-/g, '_'),
```
Comment (JST5000) [inline test/webpack.config.js:23]: What is match[0] here?

**Review by JST5000 — APPROVED** [review]
Moving this over seems like a straightforward change, although I'm still a bit confused about how the webpack.config.js works.

File: test/localIntegrationRunner.html [hunk 2]
```diff
@@ -17,7 +17,7 @@
       mocha.ui('bdd')
     </script>
 
-    <script src="../testCompiledForWeb/index.js"></script>
```
Comment (mvadari) [inline test/localIntegrationRunner.html:20]: can we put this file in `test/integration`?

File: test/webpack.config.js (hunk 1 above)
Comment (mvadari) [reply test/webpack.config.js:23]: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/match

Comment (mvadari) [reply test/webpack.config.js:23]: It's a regex thing

File: test/webpack.config.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
+'use strict'
```
Comment (mvadari) [inline test/webpack.config.js:1]: can this also go in `test/browser`?

**Review by mvadari — APPROVED** [review]
left a couple minor comments

**Comment by mvadari** [comment]
> I'm still a bit confused about how the webpack.config.js works.

Honestly, I don't really understand how webpacking works either. It's a pretty big rabbit hole.

File: test/webpack.config.js (hunk 3 above)
Comment (natenichols) [reply test/webpack.config.js:1]: It could, but it'd prolly fit better in `integration`. I'm okay with leaving it in `test` for now.

**Pull request merged**